### PR TITLE
Add CTest for XRootd Remote support

### PR DESCRIPTION
**Pull request by eisenhauer**

### CODE REVIEW

**Comment by vicentebolea** [comment]
@eisenhauer I added tmp images to check that xrootd is installed correctly

**Comment by eisenhauer** [comment]
Cool.  Lets see how this goes...

**Comment by vicentebolea** [comment]
@eisenhauer find the temp images with xrootd.

**Review by eisenhauer — COMMENTED** [review]
Digging into the XRootD code here, this error means that getpwnam("daemon") didn't return anything...  If you pass a numeric value it passes that to getpwuid().

**Comment by vicentebolea** [comment]
@eisenhauer the tests finally passes

File: testing/adios2/engine/bp/generateXRootDConfig.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+echo "First arg is $1"
+mkdir -p xroot/var/spool
+mkdir -p xroot/run/xrootd
+mkdir -p xroot/etc/xrootd
+{
+    echo "xrootd.fslib libXrdSsi.so";
+    echo ""
+    echo "all.export /home/eisen/xroot/data nolock r/w";
```
Comment (vicentebolea) [inline testing/adios2/engine/bp/generateXRootDConfig.sh:9]: we probably want to change this

**Comment by eisenhauer** [comment]
Nice!

File: testing/adios2/engine/bp/generateXRootDConfig.sh (hunk 1 above)
Comment (eisenhauer) [reply testing/adios2/engine/bp/generateXRootDConfig.sh:9]: Agreed.

**Comment by eisenhauer** [comment]
@vicentebolea Time to revive this PR and prepare to merge. I tried (badly) to update the CI to reflect changes since it was first done, but it likely needs you attention.

**Comment by vicentebolea** [comment]
do you need to re-do the docker images?

**Comment by eisenhauer** [comment]
I'm actually not sure what you did before to get these to run. Temporary docker images?  There were conflicts with master in some of the setup files that I tried to resolve, but perhaps incorrectly.

**Comment by eisenhauer** [comment]
So I guess the goal is this:  The non-CI stuff in here mostly just enables testing of the XRootD plugin.  We'd like to test on one or more platform in CI, so we'd need xrootd installed.  Doesn't have to be everywhere, but one or two places would be good.

File: testing/adios2/engine/bp/generateXRootDConfig.sh (hunk 1 above)
Comment (eisenhauer) [reply testing/adios2/engine/bp/generateXRootDConfig.sh]: Done.

**Comment by vicentebolea** [comment]
@eisenhauer the CI is ready, the 20.04 images now have xrootd.

**Comment by eisenhauer** [comment]
Minor tweak to fix server args.

File: cmake/DetectOptions.cmake [hunk 2]
```diff
@@ -583,6 +583,16 @@ elseif(ADIOS2_USE_XRootD)
 endif()
 if(XRootD_FOUND)
   set(ADIOS2_HAVE_XRootD TRUE)
+  FIND_PROGRAM( XROOTD_SERVER_BINARY xrootd
```
Comment (vicentebolea) [inline cmake/DetectOptions.cmake:586]: lower case

File: cmake/DetectOptions.cmake [hunk 3]
```diff
@@ -583,6 +583,16 @@ elseif(ADIOS2_USE_XRootD)
...
+  		HINTS
+		${XROOTD_DIR}
+		$ENV{XROOTD_DIR}
+		/usr
+		/opt/xrootd
+		PATH_SUFFIXES bin
+		)
+		message (STATUS "xrootd server binary is ${XROOTD_SERVER_BINARY}")
```
Comment (vicentebolea) [inline cmake/DetectOptions.cmake:594]:
```suggestion
```

File: testing/adios2/engine/bp/CMakeLists.txt [hunk 4]
```diff
@@ -150,6 +150,26 @@ if ((NOT WIN32) AND ADIOS2_HAVE_SST)
...
+if (ADIOS2_HAVE_XRootD)
+   macro(add_get_xrremote_tests_helper testname)
+      add_test(NAME "Remote.BP${testname}.GetXRRemote" COMMAND Test.Engine.BP.${testname}.Serial bp5)
+      set_tests_properties(Remote.BP${testname}.GetXRRemote PROPERTIES FIXTURES_REQUIRED XRServer ENVIRONMENT "DoXRootD=1")
+   endmacro()
+
+   add_test(NAME generateXRootDConfig COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/generateXRootDConfig.sh $<TARGET_FILE:adios2_xrootd>  )
+   add_test(NAME remoteXRServerSetup   COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/start_xrootd.sh ${XROOTD_SERVER_BINARY} ${CMAKE_CURRENT_BINARY_DIR})
+
+   set_tests_properties(remoteXRServerSetup         PROPERTIES FIXTURES_SETUP    XRServer)
```
Comment (vicentebolea) [inline testing/adios2/engine/bp/CMakeLists.txt:162]: Inconsistent arg indentation

File: source/adios2/toolkit/remote/XrootdRemote.cpp [hunk 5]
```diff
@@ -134,9 +134,8 @@ void myRequest::Alert(XrdSsiRespInfoMsg &aMsg)
 
     // Print what we received
     //
-    //    fprintf(XrdSsiCl::outFile, "%s@%s: Rcvd %d bytes alert: '%s'\n", rName,
-    //    GetEndPoint().c_str(),
-    //            theMsz, theMsg);
+    fprintf(XrdSsiCl::outFile, "%s@%s: Rcvd %d bytes alert: '%s'\n", rName, GetEndPoint().c_str(),
+            theMsz, theMsg);
```
Comment (vicentebolea) [inline source/adios2/toolkit/remote/XrootdRemote.cpp:138]: seems to be a old debug message (in a log file). IS this still needed?

**Review by vicentebolea — COMMENTED** [review]
Added a few comments, looks good otherwise. After addressing the comemnts that makes sense to you I can remove the 'tmp' and we can merge this.

File: source/adios2/toolkit/remote/XrootdRemote.cpp (hunk 5 above)
Comment (eisenhauer) [reply source/adios2/toolkit/remote/XrootdRemote.cpp]: Commented back out.

File: testing/adios2/engine/bp/CMakeLists.txt (hunk 4 above)
Comment (eisenhauer) [reply testing/adios2/engine/bp/CMakeLists.txt]: done

File: cmake/DetectOptions.cmake (hunk 2 above)
Comment (eisenhauer) [reply cmake/DetectOptions.cmake]: done.

**Comment by vicentebolea** [comment]
@eisenhauer we are still having errors, let me know when the CI is green

**Review by vicentebolea — APPROVED** [review]

**Comment by vicentebolea** [comment]
@eisenhauer lets squash-merge to get rid of all this fixup commits

**Review by vicentebolea — APPROVED** [review]

**Comment by vicentebolea** [comment]
@eisenhauer images are pushed (without the tmp)

**Pull request merged**